src/page/Products.js

```diff
@@ -0,0 +1,81 @@
+import ProductGrid from '../widget/ProductGrid';
+import products from 'mocks/en-us/products.json';
+import categories from 'mocks/en-us/product-categories.json';
+import useProductsWithCategories from '../utils/hooks/useProductsWithCategory';
+import { Fragment, useState } from 'react';
+import styled from 'styled-components';
+import { css } from 'styled-components';
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: space-evenly;
+`;
+
+const CategorySection = styled.section`
+    flex: 1 1 20%;
+    min-width: 240px;
+`;
+
+const CategoryContainer = styled.fieldset`
+    border: none;
+`;
+
+const ProductSection = styled.section`
+    flex: 4 1 80%;
+    box-sizing: border-box;
+    padding-bottom: 1em;
+`;
+
+const Category = styled.label`
+    ${({active}) => active && css`    
+    font-weight: bold;
+    `}
+`;
+
+export default function Products() {
+    const fullProducts = useProductsWithCategories(products.results, categories.results);
+    const [activeCategories, setActiveCategories] = useState(() => new Set());
+
+    const handleChecked = event => {
+        if (event.target.checked) {
+            activeCategories.add(event.target.value);
+        } else {
+            activeCategories.delete(event.target.value);
+        }
+        setActiveCategories(new Set(activeCategories));
+    };
```

Comment:
Creo que aquí podrías destructurar `event` en el parámetro de la función para que puedas acceder al checked y al value más fácil. Algo así:

```suggestion
    const handleChecked = ({ target: { checked, value} }) => {
        if (checked) {
            activeCategories.add(value);
        } else {
            activeCategories.delete(value);
        }
        setActiveCategories(new Set(activeCategories));
    };
```